guides/source/association_basics.md

```diff
@@ -355,2288 +439,2469 @@ class CreateAccountHistories < ActiveRecord::Migration[7.2]
+allowing STI to function correctly with the custom column name.
+
+### Disabling the inheritance column
+
+There may be cases (like when working with a legacy database) where you need to
+disable Single Table Inheritance altogether. If you don't disable STI properly,
+you might encounter an [`ActiveRecord::SubclassNotFound`][] error.
 
-While Rails uses intelligent defaults that will work well in most situations, there may be times when you want to customize the behavior of the `has_many` association reference. Such customizations can easily be accomplished by passing options when you create the association. For example, this association uses two such options:
+To disable STI, you can set the [inheritance_column][] to `nil`.
 
 ```ruby
-class Author < ApplicationRecord
-  has_many :books, dependent: :delete_all, validate: false
+# Schema: vehicles[ id, type, created_at, updated_at ]
+class Vehicle < ApplicationRecord
+  self.inheritance_column = nil
 end
-```
 
-The [`has_many`][] association supports these options:
+Vehicle.create!(type: "Car")
+# => #<Vehicle type: "Car", color: "Red", price: 10000>
+```
 
-* `:as`
-* `:autosave`
-* `:class_name`
-* `:counter_cache`
-* `:dependent`
-* `:disable_joins`
-* `:ensuring_owner_was`
-* `:extend`
-* `:foreign_key`
-* `:foreign_type`
-* `:inverse_of`
-* `:primary_key`
-* `:query_constraints`
-* `:source`
-* `:source_type`
-* `:strict_loading`
-* `:through`
-* `:validate`
+In this configuration, Rails will treat the type column as a normal attribute
+and will not use it for STI purposes. This is useful if you need to work with a
+legacy schema that does not follow the STI pattern.
 
-##### `:as`
+These adjustments provide flexibility when integrating Rails with existing
+databases or when specific customization is required for your models.
 
-Setting the `:as` option indicates that this is a polymorphic association, as discussed [earlier in this guide](#polymorphic-associations).
+[inheritance_column]:
+    https://api.rubyonrails.org/classes/ActiveRecord/ModelSchema.html#method-c-inheritance_column
+[`ActiveRecord::SubclassNotFound`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/SubclassNotFound.html
 
-##### `:autosave`
+Delegated Types
+----------------
 
-If you set the `:autosave` option to `true`, Rails will save any loaded association members and destroy members that are marked for destruction whenever you save the parent object. Setting `:autosave` to `false` is not the same as not setting the `:autosave` option. If the `:autosave` option is not present, then new associated objects will be saved, but updated associated objects will not be saved.
+[`Single Table Inheritance (STI)`](#single-table-inheritance-sti) works best
+when there is little difference between subclasses and their attributes, but it
+includes all attributes of all subclasses in a single table.
 
-##### `:class_name`
+The disadvantage of this approach is that it can result in table bloat, as the
```

Comment:
It would be helpful to mention the lack of referential integrity as another disadvantage.

Reply:
@mstroming thanks - I've added a [considerations section](https://github.com/rails/rails/blob/2a8a870deb4b79cb310619fdc304d89e9e462a3d/guides/source/association_basics.md#considerations) to STI to maintain referential integrity.